### PR TITLE
Remove UserMenu component render modifier

### PR DESCRIPTION
**Pull request by michaelchadwick**
Refs ilios/ilios#5374

Also fixed a bug where if you keyboard tabbed over to the user menu and pressed UpArrow, it would throw a console error: `Uncaught TypeError: Cannot read properties of undefined (reading 'previousElementSibling')`

### CODE REVIEW

**Comment by michaelchadwick** [comment]
Added keyboard navigation testing to `UserMenu` component. Needed to add a slight delay in both the component and integration test to properly set focus and test for it.

File: packages/frontend/app/components/user-menu.js [hunk 1]
```diff
@@ -21,6 +19,12 @@ export default class UserMenuComponent extends Component {
   @action
   toggleMenu() {
     this.isOpen = !this.isOpen;
+
+    if (this.isOpen) {
+      setTimeout(() => {
```
Comment (jrjohnson) [inline packages/frontend/app/components/user-menu.js:24]: I'm guessing this is the same thing the `afterRender` runloop was handling as well. A timeout here is doing the same thing, moving execution out of the current thread, waiting for all things to finish (which includes the rendering) and then doing it on the next pass through. I'd prefer to either see the runloop used here, as that's what it's really for, or convert this into a `task` and then use the `await timeout(1)` syntax. That's doing the exact same thing as your timeout here, but is more idiomatic for our codebase.

File: packages/frontend/tests/integration/components/user-menu-test.js [hunk 2]
```diff
@@ -11,6 +11,9 @@ module('Integration | Component | user-menu', function (hooks) {
   setupRenderingTest(hooks);
   setupMirage(hooks);
 
+  // My Profile and Logout
+  const linkCount = 2;
```
Comment (jrjohnson) [inline packages/frontend/tests/integration/components/user-menu-test.js:15]: Nice change for clarity!

File: packages/frontend/tests/integration/components/user-menu-test.js [hunk 3]
```diff
@@ -56,8 +59,62 @@ module('Integration | Component | user-menu', function (hooks) {
     await render(hbs`<UserMenu />`);
 
     await component.toggle.down();
-    assert.strictEqual(component.links.length, 2);
+    assert.strictEqual(component.links.length, linkCount);
+    await component.toggle.click();
+    assert.strictEqual(component.links.length, 0);
+  });
+
+  test('keyboard navigation', async function (assert) {
```
Comment (jrjohnson) [inline packages/frontend/tests/integration/components/user-menu-test.js:67]: 🔥

File: packages/frontend/app/components/user-menu.js [hunk 4]
```diff
@@ -49,30 +53,29 @@ export default class UserMenuComponent extends Component {
...
       }
     }
   }
 
   handleArrowUp(item) {
-    if (item.previousElementSibling) {
-      item.previousElementSibling.querySelector('a').focus();
-    } else {
-      this.element.querySelector('.menu li:last-of-type a').focus();
+    if (item) {
```
Comment (jrjohnson) [inline packages/frontend/app/components/user-menu.js:73]: I'm ok with this. Could also use the `item?.previousIndex...` optional chaining syntax which would do the same thing with slightly less indentation.

**Review by jrjohnson — CHANGES_REQUESTED** [review]

File: packages/frontend/app/components/user-menu.js (hunk 4 above)
Comment (michaelchadwick) [reply packages/frontend/app/components/user-menu.js:73]: I didn't like it when I wrote it, so thank you for reminding me of the `?` optional thing!

File: packages/frontend/app/components/user-menu.js (hunk 1 above)
Comment (michaelchadwick) [reply packages/frontend/app/components/user-menu.js]: Initially, I just fixed this by going the runloop path, but we have to use an `eslint-disable` to do that, which seems bad and only makes the code worse. So, I turned it into a `task` as you suggested.

File: packages/frontend/app/components/user-menu.js [hunk 5]
```diff
@@ -49,30 +57,24 @@ export default class UserMenuComponent extends Component {
...
     }
   }
 
   handleArrowUp(item) {
-    if (item.previousElementSibling) {
+    if (item?.previousElementSibling) {
       item.previousElementSibling.querySelector('a').focus();
     } else {
-      this.element.querySelector('.menu li:last-of-type a').focus();
+      item.parentElement.lastElementChild.querySelector('a').focus();
```
Comment (jrjohnson) [inline packages/frontend/app/components/user-menu.js:77]:
```suggestion
      item?.parentElement.lastElementChild.querySelector('a').focus();
```

**Review by jrjohnson — COMMENTED** [review]
I *think* there needs to be another check on item?

**Review by jrjohnson — APPROVED** [review]

**Pull request merged**